spec/models/reservations_validations_spec.rb

```diff
@@ -24,4 +24,40 @@
     reservation.update_attributes reserve_start_at: now, reserve_end_at: now + 1.hour + 5.minutes
     expect(reservation.errors[:base]).to be_present
   end
+
+  describe "starts_before_cutoff" do
+    context "when there is a cutoff" do
+      before :each do
+        reservation.product.update_attribute(:cutoff_hours, 2)
+      end
+
+      context "when reservation is after the cutoff" do
+        let(:reservation) { build :setup_reservation, reserve_start_at: Time.zone.now + 3.hours }
+        let(:user) { reservation.order_detail.created_by_user }
+
+        it "saves the reservation" do
+          expect(reservation.save_as_user(user)).to eq(true)
```

Comment:
Minor (and not show-stopping): this project goes more for `be true` over `eq(true)` style.
Though in this particular case I might have even considered `expect { reservation.save_as_user!(user) }.not_to raise_error`